src/components/SelectedSurveyPage.jsx

```diff
@@ -26,6 +26,19 @@ export default class SelectedSurveyPage extends PureComponent {
     fetchSurvey(id);
   }
 
+  // TODO(ayushbhagat): Check for cycles.
```

Comment:
Good point. Yeah, I think we should check for cycles in the backend as well. @chrisjluc @dinahshi